frontend/src/Frontend/UI/Widgets.hs

```diff
@@ -878,29 +883,39 @@ uiGasPriceInputField eReset conf = dimensionalInputFeedbackWrapper (Just "KDA")
 
 -- | Let the user pick a chain id.
 userChainIdSelect
-  :: (MonadWidget t m, HasNetwork model t
+  :: ( MonadWidget t m
+     , HasNetwork model t
      )
   => model
-  -> m (MDynamic t ChainId)
+  -> m ( Dropdown t (Maybe ChainId)
+       , MDynamic t ChainId
+       )
 userChainIdSelect m =
-  userChainIdSelectWithPreselect m (constDyn Nothing)
+  userChainIdSelectWithPreselect m True (constDyn Nothing)
 
 -- | Let the user pick a chain id but preselect a value
 userChainIdSelectWithPreselect
   :: (MonadWidget t m, HasNetwork model t
      )
   => model
+  -> Bool
   -> Dynamic t (Maybe ChainId)
-  -> m (MDynamic t ChainId)
-userChainIdSelectWithPreselect m mChainId = mkLabeledClsInput True "Chain ID" (uiChainSelection mNodeInfo mChainId)
-  where mNodeInfo = (^? to rights . _head) <$> m ^. network_selectedNodes
+  -> m ( Dropdown t (Maybe ChainId)
+       , MDynamic t ChainId
+       )
+userChainIdSelectWithPreselect m inlineLabel mChainId =
+  mkLabeledClsInput inlineLabel "Chain ID" (uiChainSelection mNodeInfo mChainId)
+  where
+    mNodeInfo = (^? to rights . _head) <$> m ^. network_selectedNodes
 
 uiChainSelection
   :: MonadWidget t m
   => Dynamic t (Maybe NodeInfo)
   -> Dynamic t (Maybe ChainId)
   -> CssClass
-  -> m (Dynamic t (Maybe ChainId))
+  -> m ( Dropdown t (Maybe ChainId)
+       , Dynamic t (Maybe ChainId)
```

Comment:
Unlike when we return `(Event t, Dynamic t)`, these are redundant, but callers don't know that. I'd just return `Dropdown t`.